hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -279,15 +279,37 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 
 	// Set Ignition Server endpoint
 	{
+		serviceStrategy := servicePublishingStrategyByType(hcluster, hyperv1.Ignition)
+		if serviceStrategy == nil {
```

Comment:
Although structural schema enum validation should prevent this from ever be the case I wonder in addition to logs should we also set a condition that signal this in a more ux friendly manner?

I wonder should we log the error but not return it here? since this won't get fixed by requeueing unless there's and update to the API which will already emit an update event a reconciliation will happen.

Reply:
I think we need to return error as it doesn't make sense to continue execution if this has failed. However I could also log if that helps.

Reply:
Yeh we need to break execution but we don't necessarily need to return an error.
If we return error the controller runtime implementation will automatically requeue the event again and again. In this case we know that this particular problem will never be fixed by requeueing because the problem is a bad input.
So I wonder if instead of returning the error we want to just to log it and return nil.
Basically this:
https://github.com/openshift/hypershift/blob/b7566016fc3a287435f32d625abdf447e458e301/hypershift-operator/controllers/nodepool/nodepool_controller.go#L243-L246

Reply:
OHHHHH I see what you mean

Reply:
This is neat sorry no that makes sense......

Reply:
I will change to this I agree pointless to auto requeue

Reply:
fixed